thrust/thrust/complex.h

```diff
@@ -76,7 +76,7 @@ namespace detail
 template <typename T, std::size_t Align>
 struct complex_storage;
 
-#if THRUST_CPP_DIALECT >= 2011 && (THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_GCC) && (THRUST_GCC_VERSION >= 40800)
+#if THRUST_CPP_DIALECT >= 2011 && defined(_CCCL_COMPILER_GCC) && (_CCCL_GCC_VERSION >= 40800)
```

Comment:
Nitpick: This should be either
```suggestion
#if defined(_CCCL_COMPILER_GCC) && (_CCCL_GCC_VERSION >= 40800)
```

or
```suggestion
#if _CCCL_STD_VER >= 2011 && defined(_CCCL_COMPILER_GCC) && (_CCCL_GCC_VERSION >= 40800)
```

Reply:
I went for a slighly bigger refactoring and dropped everything for the plain C++11 implementation.